addons/website/static/src/snippets/s_facebook_page/options.js

```diff
@@ -121,7 +121,8 @@ options.registry.facebookPage = options.Class.extend({
                     return this.fbData.tabs.split(',').includes(optionName.replace(/^tab./, ''));
                 } else {
                     if (optionName === 'show_cover') {
-                        return !this.fbData.hide_cover;
+                        // Sometimes a string, sometimes a boolean.
+                        return String(this.fbData.hide_cover) === "false";
```

Comment:
Isn't it a boolean internally in the JS code only because the `defaults` is setting it as a boolean?
Which is is actually wrong too as it's not replicating how it is stored in the DOM, right?

https://github.com/odoo/odoo/blob/03c552690b15cbf2e7d6b7812386ac64042219af/addons/website/static/src/snippets/s_facebook_page/options.js#L15-L23

Those `: false` and `= true` should be `: "false"` and `: "true"` imo?

Reply:
Not only, it is also set to a boolean by `toggleOption`: https://github.com/odoo/odoo/blob/03c552690b15cbf2e7d6b7812386ac64042219af/addons/website/static/src/snippets/s_facebook_page/options.js#L72 
I could replace this by `widgetValue ? "false": "true"` but that feels even worse.

Reply:
/Summon @qsm-odoo You would advice better than I would here

Reply:
@rdeodoo I don't know, whatever :shrug: Probably:
- In stable, as @bso-odoo did, that way it works no matter what
- In master, adapt the code to either always work with a boolean or always with a string, whatever leads to the less code I guess (if the same, then a real boolean would be prettier? :shrug:)